### PR TITLE
Add run time command switch to specify the mpas-bundle build directory

### PR DESCRIPTION
**Pull request by jim-p-w**
Add run time command switch to specify the mpas-bundle build directory.
Also add a run time switch to specify a scenario name suffix

Output of running `./Run.py -h`:
```
usage: Run.py [-h] [-b BUNDLE_DIR] [-x SUFFIX] config

positional arguments:
  config                configuration file; e.g., runs/test.yaml, scenarios/{{scenarioName}}.yaml

options:
  -h, --help            show this help message and exit
  -b BUNDLE_DIR, --bundle_dir BUNDLE_DIR
                        mpas bundle directory
  -x SUFFIX, --suffix SUFFIX
                        experiment name suffix
```
Note this change is backward compatible, `Run.py <config>` works as it used to.

#### Tier 1:
 - [x] 3dvar_OIE120km_WarmStart
 - [x] 3denvar_OIE120km_IAU_WarmStart
 - [x] 3dvar_OIE120km_ColdStart
 - [x] 3dvar_O30kmIE60km_ColdStart
 - [x] 3denvar_O30kmIE60km_WarmStart
 - [x] eda_OIE120km_WarmStart
 - [x] getkf_OIE120km_WarmStart
 - [x] ForecastFromGFSAnalysesMPT
 - [x] 4denvar_OIE120km_WarmStart
 - [x] 4dhybrid_OIE120km_WarmStart_

The following tests failed when specifying an mpas-bundle build of the develop branch:
  - 3dvar_O30kmIE60km_ColdStart
  - 3dvar_OIE120km_ColdStart
  
#### Tier 2 (optional):
 - [ ] GenerateGFSAnalyses
 - [ ] GenerateObs

### CODE REVIEW

File: initialize/framework/Build.py [hunk 1]
```diff
@@ -40,7 +40,10 @@ def __init__(self, config:Config, model:Model=None):
     # set system dependent defaults before invoking Component ctor
     system = os.getenv('NCAR_HOST')
     if system == 'derecho':
-      self.variablesWithDefaults['mpas bundle'] = \
+      if config._bundle_dir != None:
+        self.variablesWithDefaults['mpas bundle'] = [config._bundle_dir, str]
+      else:
+        self.variablesWithDefaults['mpas bundle'] = \
           ['/glade/campaign/mmm/parc/ivette/pandac/codeBuild/mpasBundle_dev_SP/mpas-bundle/build', str]
```
Comment (amstokely) [inline initialize/framework/Build.py]: I don't think it's a great idea to have hard coded file paths in the source code. If that directory ever changes/gets deleted then it breaks the entire library.

**Comment by liujake** [comment]
I do not think making mpas-bundle location configurable very useful. Actually we have a lot of other 'location' things to configure for different experiments: obs input, varbc locations, ensemble input, static B files, model namelist/stream, DA namelist/yaml etc. Currently those settings are distributed in different yaml files. IMO, to ease porting the workflow to a different machine or configure different experiments on the same machine, collecting those 'location' settings in a single file will be helpful. Eventually, our top 'scenario' yaml file for a single experiment can have all those locations explicitly set I believe, but most of time, people will use a combination of default localizations and some in top scenario settings. One example file to collect all localition settings in one single file would be useful.

**Comment by jim-p-w** [comment]
> I do not think making mpas-bundle location configurable very useful. Actually we have a lot of other 'location' things to configure for different experiments: obs input, varbc locations, ensemble input, static B files, model namelist/stream, DA namelist/yaml etc. Currently those settings are distributed in different yaml files. IMO, to ease porting the workflow to a different machine or configure different experiments on the same machine, collecting those 'location' settings in a single file will be helpful. Eventually, our top 'scenario' yaml file for a single experiment can have all those locations explicitly set I believe, but most of time, people will use a combination of default localizations and some in top scenario settings. One example file to collect all localition settings in one single file would be useful.

I did this specifically so I could point to weekly builds of the develop branch of mpas-bundle from cron jobs. Ditto for adding a suffix to a scenario, so I could automate generating comparison graphs from one week's build to the next weeks build. I'd be happy to have a conversation about this.

**Comment by liujake** [comment]
For this special use case, it is good.

**Comment by liujake** [comment]
@jim-p-w Can you resolve the conflict first?

**Comment by jim-p-w** [comment]
> @jim-p-w Can you resolve the conflict first?

@liujake conflict is resolved.

**Review by ibanos90 — APPROVED** [review]
Tested the branch and it worked as expected, i.e., it used the mpas bundle specified via command line. Thanks @jim-p-w!

**Comment by liujake** [comment]
@jim-p-w may merge given you have an approval.

**Pull request merged**